Source/SteveSloth/Public/MyEnemy.h

```diff
@@ -0,0 +1,41 @@
+/****************************************************************************************
+Copyright: SteveSloth
+Name: Aidan
+Script: MyEnemy.h
+Date: April 23. 2024
+Description: This is the base class for all Enemies.
+TODO:
+Known Bugs:
+TODO:
+Known Bugs:
+****************************************************************************************/
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "GameFramework/Character.h"
+#include "MyEnemy.generated.h"
+
+/**
+ * 
+ */
+
+enum MovementState {Idle, Patrolling, Chasing, Fleeing, Walking};
+enum AttackType {Melee, Ranged};
+enum Level {One, Two, Three, Four, Five};
+
+class STEVESLOTH_API MyEnemy
+{
+public:
+	float MaxHealth;
+	float CurrentHealth;
+	int Damage;
+
+	float MoveSpeed;
+	bool IsDead;
+
```

Comment:
Is there a reason why all these variables are public? if anything they should be private or protected, i would lean more towards protected seeing as how this is a base class that all enemies will inherit from

Reply:
All the variables should be protected as All enemies will be using them for their own values